core/vm/evm.go

```diff
@@ -352,8 +352,20 @@ func (evm *EVM) StaticCall(caller ContractRef, addr common.Address, input []byte
 	return ret, contract.Gas, err
 }
 
+type codeAndHash struct {
+	code []byte
+	hash common.Hash
+}
+
+func (c *codeAndHash) Hash() common.Hash {
+	if c.hash == emptyCodeHash {
```

Comment:
This is not correct AFAIK. It should be `common.Hash{}`. The `emptyCodeHash` is `Keccak256("")`.